wavecrestweb/kenworthy-machine
# Km 23/feature machine quote form

## Description
This PR adds a submit a quote request form to the machines page as well as a success message on successful submit. The form title and 2 of the field placeholders are connected to TinaCMS. It also refactors the components directory to organize components by page.


## Related Tickets & Documents

Closes #KM-23 

## Acceptance Criteria



## What type of PR is this? (check all applicable)


|     | Type                       |
| --- | -------------------------- |
| x | :sparkles: New feature     |
| x| 🎨 Style                   |
|  x | :hammer: Refactoring       |

## Mobile & Desktop Screenshots/Recordings

<img width="1418" alt="Screenshot 2024-03-05 at 12 11 55 PM" src="https://github.com/wavecrestweb/kenworthy-machine/assets/115492619/dca26783-2fac-41e5-8936-2d079322ed6f">


## What I learned
I definitely tried to over-engineer this solution by conditionally rendering the same form component based on which page was rendering it. However, since the layout is different I finally realized it was simpler to create a new form. I did create a custom hook to separate and simplify the form submission logic.

## Testing steps

1.  `npm switch KM-23/feature-machine-quote-form` (and `npm i` if you haven't for the previous form PR)
2.  `npm run dev`
3.  navigate to the machines page and try to submit the form
4.  upon successful submit try to navigate back to the form to submit again
5.  open TinaCMS and see if form fields and title are editable
6.  compare the design/layout to the wireframe


## Added to documentation?

|     | Type                       |
| --- | -------------------------- |
|     | 📜 README.md    |
|  ✓ | 🙅 no documentation needed    |


## What gif best describes this PR?
 ![overcomplicated](https://media.giphy.com/media/iF8IaDx2N6vfzS2k52/giphy.gif)
